src/figuresClasses.ts

```diff
@@ -1,11 +1,85 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+function checkOnError(errorMessage: string, ...sides: number[]): void {
+  if (sides.some((side) => side <= 0)) {
+    throw new Error(errorMessage);
+  }
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const triangleErrorMessage =
+      'Invalid triangle sides: ensure all sides are positive.';
```

Comment:
The error message for an invalid triangle could be more descriptive. While it does mention that all sides must be positive, it could also include information about the triangle inequality theorem to be clearer.